angular-chart.js

```diff
@@ -1,243 +1,302 @@
+    .factory('ChartJsFactory', ['ChartJs', ChartJsFactory])
+    .directive('chartBase',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory(); }])
+    .directive('chartLine',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('Line'); }])
+    .directive('chartBar',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('Bar'); }])
+    .directive('chartRadar',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('Radar'); }])
+    .directive('chartDoughnut',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('Doughnut'); }])
+    .directive('chartPie',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('Pie'); }])
+    .directive('chartPolarArea',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('PolarArea'); }]);
+
+  /**
+   * Wrapper for chart.js
+   * Allows configuring chart js using the provider
+   *
+   * angular.module('myModule', ['chart.js']).config(function(ChartJsProvider) {
+   *   ChartJsProvider.setOptions({ responsive: true });
+   *   ChartJsProvider.setOptions('Line', { responsive: false });
+   * })))
+   */
+  function ChartJsProvider() {
+    var Chart = (typeof window !== 'undefined' && window.Chart) ||
+      (typeof require !== 'undefined' && require('chart.js'));
+
+    if (typeof Chart !== 'function') {
+      throw 'Failed to load chart.js';
+    }
 
-        // Order of setting "watch" matter
+    var ChartJs = {
+      Chart: Chart,
+      options: {
+        responsive: true,
+        multiTooltipTemplate: '<%if (datasetLabel){%><%=datasetLabel%>: <%}%><%= value %>',
+        colours: [
+          '#97BBCD', // blue
+          '#DCDCDC', // light grey
+          '#F7464A', // red
+          '#46BFBD', // green
+          '#FDB45C', // yellow
+          '#949FB1', // grey
+          '#4D5360'  // dark grey
+        ]
+      }
+    };
+
+    /**
+     * Allow to set global options during configuration
+     */
+    this.setOptions = function (type, options) {
+      // If no type was specified set option for the global object
+      if (options === undefined) {
+        options = type;
+        ChartJs.options = angular.extend(ChartJs.options, options);
+        return;
+      }
+      // Set options for the specific chart
+      ChartJs.options[type] = angular.extend(ChartJs.options[type] || {}, options);
+    };
 
-        scope.$watch('data', function (newVal, oldVal) {
-          if (! newVal || ! newVal.length || (Array.isArray(newVal[0]) && ! newVal[0].length)) return;
-          var chartType = type || scope.chartType;
-          if (! chartType) return;
+    this.$get = function () {
+      return ChartJs;
+    };
+  }
 
-          if (chart) {
-            if (canUpdateChart(newVal, oldVal)) return updateChart(chart, newVal, scope);
-            chart.destroy();
+  function ChartJsFactory(ChartJs) {
```

Comment:
Minor nitpick, add a space before round braces in function declarations (I need to add the corresponding `jscs` rule).